src/main/java/org/itsallcode/openfasttrace/importer/SpecificationListBuilder.java

```diff
@@ -166,22 +178,17 @@ public void setLocation(final Location location)
     @Override
     public void endSpecificationItem()
     {
-        // [impl->dsn~ignoring-artifact-types-during-import~1]
-        if (this.itemBuilder != null && !ignoreCurrentArtifact())
+        // [impl->dsn~filtering-by-artifact-types-during-import~1]
+        if (this.itemBuilder != null && isAccepted(this.id.getArtifactType()))
         {
             createNewSpecificationItem();
         }
     }
 
-    private boolean ignoreCurrentArtifact()
-    {
-        return isIgnoredArtifact(this.id.getArtifactType());
-    }
-
-    private boolean isIgnoredArtifact(final String artifactType)
+    private boolean isAccepted(final String artifactType)
     {
-        return this.ignoredArtifactTypes != null
-                && this.ignoredArtifactTypes.contains(artifactType);
+        return !this.filterSettings.isArtifactTypeCriteriaSet()
+                || this.filterSettings.getArtifactTypes().contains(artifactType);
```

Comment:
Make sure that `getArtifactTypes()` returns a `Set` to improve lookup performance